### ISSUE TITLE
OM2 Units & Scales

### ISSUE THREAD

**Issue by aanastasiou** [HajoRijgersberg/OM#35]
This is a continuation of the discussion taking place in [this](https://github.com/HajoRijgersberg/OM/issues/7#issuecomment-641420345) ticket.

The main question there was: "Would it be a good practice to define generic scales so that certain computational features do get a scale or is it better to leave them without a unit?". For example, the value of an image feature may be taking values within the range of 0..255 and therefore could be represented in a generic uint8 scale.

**Comment by HajoRijgersberg** [comment]
Thanks, Athanasios!
A very good question. In principle, I want an ontology to meet needs. Those needs must be accepted. For example, we indicate an image feature in practice without unit or scale. Perhaps not ideal, but apparently not a problem for most purposes, especially with unitless quantities such as numbers and fractions. So I would like the property om:hasValue of the class Quantity to have as range not only om:Measure and om:Point (the first is a construct like '3 kilogram' - so a number with a unit - and the second is a point on a scale, such as the melting point of water that is defined for spanning the Celsius scale in OM), but also Float. The problem is that, in RDF, this property will then both be an object property and a datatype property. Anyway, that concerns technical implementation, which anticipates this fundamental discussion.

**Comment by aanastasiou** [comment]
@HajoRijgersberg Yes. I intend to close this ticket at some point but in the meantime we might have to treat this as a discussion forum rather than a problem-response repository.

Yes, I understand and that is fine.

Just a point on Quantity. You refer to its hasValue property. However, that is not obvious from the UML diagram. Perhaps it is a property that "trickles down" from higher up in the lineage of Quantity. In the ontology itself, when you load the ontology, it is there. But not on the diagram. 

Is there an updated diagram or did I miss a branch somewhere? I suppose hasValue expects a Measure at its end (?).

**Comment by HajoRijgersberg** [comment]
Yes, discussion is fine! :)

Indeed, the property is not in Figure 1. I should include it. I didn't because the hasValue property relates to  the field of statements. It is in some way something like an equation, like _a_ = 3 m. My idea was that that field should be worked out in a different ontology, which uses OM. But a need was expressed for that property, so I included it.
In Figure 2 by the way the property is included.

Indeed hasValue has Measure as its range and Point (of a scale).

**Comment by aanastasiou** [comment]
@HajoRijgersberg Can I please ask what is `dimensionOne` (?) Is it related to "unit one" that came up in our earlier discussion?

**Comment by HajoRijgersberg** [comment]
Dimension one is indeed the dimension that is related to unit one. The dimension has for all base dimensions (i.e., length dimension, mass dimension, time dimension, etc.) the exponent 0.

**Comment by aanastasiou** [comment]
@HajoRijgersberg I see, so this is the "default" dimension that a potential Feature class could be associated with. Correct?

**Comment by HajoRijgersberg** [comment]
Yes, correct! :)
